src/main/java/org/trigon/hardware/rev/spark/io/SimulationSparkIO.java

```diff
@@ -63,51 +77,52 @@ public void setInverted(boolean inverted) {
+    public void setBrake(boolean brake) {
+        final SparkMaxConfig configuration = new SparkMaxConfig();
+        configuration.idleMode(brake ? SparkMaxConfig.IdleMode.kBrake : SparkMaxConfig.IdleMode.kCoast);
+        motor.configure(configuration, SparkBase.ResetMode.kNoResetSafeParameters, SparkBase.PersistMode.kNoPersistParameters);
     }
 
     @Override
-    public void setClosedLoopRampRate(double rampRate) {
-        motor.setClosedLoopRampRate(rampRate);
+    public void updateSimulation() {
+        if (physicsSimulation == null)
+            return;
+
+        updatePhysicsSimulation();
+        updateMotorSimulation();
+        updateEncoderSimulation();
     }
 
     @Override
-    public void setSmartCurrentLimit(int limit) {
-        motor.setSmartCurrentLimit(limit);
+    public void configure(SparkBaseConfig configuration, SparkBase.ResetMode resetMode, SparkBase.PersistMode persistMode) {
+        motor.configure(configuration, resetMode, persistMode);
     }
 
     @Override
-    public void setOpenLoopRampRate(double rampRate) {
-        motor.setOpenLoopRampRate(rampRate);
+    public void setPhysicsSimulation(MotorPhysicsSimulation physicsSimulation, boolean isUsingAbsoluteEncoder) {
+        this.physicsSimulation = physicsSimulation;
+        this.isUsingAbsoluteEncoder = isUsingAbsoluteEncoder;
     }
 
-    @Override
-    public void setPID(double p, double i, double d) {
-        pidController.setP(p);
-        pidController.setI(i);
-        pidController.setD(d);
+    private boolean isUsingAbsoluteEncoder() {
+        return isUsingAbsoluteEncoder;
     }
 
-    @Override
-    public void setConversionsFactor(double conversionsFactor) {
-        encoder.setConversionsFactor(conversionsFactor);
+    private void updatePhysicsSimulation() {
+        physicsSimulation.setInputVoltage(motorSimulation.getBusVoltage() * motorSimulation.getAppliedOutput());
+        physicsSimulation.updateMotor();
     }
 
-    @Override
-    public void restoreFactoryDefaults() {
-        motor.restoreFactoryDefaults();
+    private void updateMotorSimulation() {
+        motorSimulation.iterate(getConversionFactor() * (isUsingAbsoluteEncoder ? Conversions.perMinuteToPerSecond(physicsSimulation.getSystemVelocityRotationsPerSecond()) : Conversions.perMinuteToPerSecond(physicsSimulation.getRotorVelocityRotationsPerSecond())), RobotHardwareStats.SUPPLY_VOLTAGE, RobotHardwareStats.getPeriodicTimeSeconds());
+        motorSimulation.setMotorCurrent(physicsSimulation.getCurrent());
     }
 
-    @Override
-    public void burnFlash() {
-        motor.burnFlash();
+    private void updateEncoderSimulation() {
+        absoluteEncoderSimulation.iterate(getConversionFactor() * (isUsingAbsoluteEncoder() ? Conversions.perMinuteToPerSecond(physicsSimulation.getSystemVelocityRotationsPerSecond()) : Conversions.perMinuteToPerSecond(physicsSimulation.getRotorVelocityRotationsPerSecond())), RobotHardwareStats.getPeriodicTimeSeconds());
```

Comment:
the ternary operator makes this very unreadable. extract. 
also, calculating this twice isn't efficient.